vtm/src/org/oscim/layers/tile/bitmap/BitmapTileLayer.java

```diff
@@ -55,13 +68,21 @@ public FadeStep(int zoomStart, int zoomEnd, float alphaStart, float alphaEnd) {
         }
 
         public FadeStep(double scaleStart, double scaleEnd, float alphaStart, float alphaEnd) {
+            if (scaleStart > scaleEnd)
+                throw new IllegalArgumentException("scaleStart must be smaller than scaleEnd");
             this.scaleStart = scaleStart;
             this.scaleEnd = scaleEnd;
             this.zoomStart = Math.log(scaleStart) / Math.log(2);
             this.zoomEnd = Math.log(scaleEnd) / Math.log(2);
             this.alphaStart = alphaStart;
             this.alphaEnd = alphaEnd;
         }
+
+        @Override
+        public int compareTo(FadeStep o) {
+            if (o == null) return 1;
+            return (int) (this.zoomStart - o.zoomStart);
+        }
     }
```

Comment:
- How a `null` can produce a valid comparison?
- Can use `Integer.compare` to compare `int`s.

Reply:
- thought everything is greater than `null`, but can remove it as it should not be compared.
- ok